engine/runtime/action-processor/src/wasm/runtime_executor.rs

```diff
@@ -0,0 +1,277 @@
+        vec![DEFAULT_PORT.clone()]
+    }
+
+    fn get_output_ports(&self) -> Vec<Port> {
+        vec![DEFAULT_PORT.clone()]
+    }
+
+    fn build(
+        &self,
+        _ctx: NodeContext,
+        _event_hub: EventHub,
+        _action: String,
+        with: Option<HashMap<String, Value>>,
+    ) -> Result<Box<dyn Processor>, BoxedError> {
+        let params: WasmRuntimeExecutorParam = if let Some(with) = with {
+            let value: Value = serde_json::to_value(with).map_err(|e| {
+                WasmProcessorError::RuntimeExecutorFactory(format!(
+                    "Failed to serialize `with` parameter: {}",
+                    e
+                ))
+            })?;
+            serde_json::from_value(value).map_err(|e| {
+                WasmProcessorError::RuntimeExecutorFactory(format!(
+                    "Failed to deserialize `with` parameter: {}",
+                    e
+                ))
+            })?
+        } else {
+            return Err(WasmProcessorError::RuntimeExecutorFactory(
+                "Missing required parameter `with`".to_string(),
+            )
+            .into());
+        };
+
+        self.compile_to_wasm(&params)?;
+
+        let process = WasmRuntimeExecutor { params };
+        Ok(Box::new(process))
+    }
+}
+
+impl WasmRuntimeExecutorFactory {
+    fn compile_to_wasm(&self, params: &WasmRuntimeExecutorParam) -> super::errors::Result<()> {
+        match params.programming_language {
+            ProgrammingLanguage::Python => {
+                let py2wasm_output = std::process::Command::new("py2wasm")
+                    .args([&params.source_code_file_path, "-o", &params.wasm_file_path])
+                    .output()
+                    .map_err(|e| {
+                        WasmProcessorError::RuntimeExecutorFactory(format!(
+                            "Failed to run py2wasm: {}. Command: py2wasm {} -o {}",
+                            e, params.source_code_file_path, params.wasm_file_path
+                        ))
+                    })?;
+
+                if !py2wasm_output.status.success() {
+                    let error_msg = String::from_utf8_lossy(&py2wasm_output.stderr);
+                    return Err(WasmProcessorError::RuntimeExecutorFactory(format!(
+                        "Python compilation failed: {}",
+                        error_msg
+                    )));
+                }
+            }
+            _ => {
+                return Err(WasmProcessorError::RuntimeExecutorFactory(
+                    "Unsupported programming language".to_string(),
+                ));
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct WasmRuntimeExecutor {
+    params: WasmRuntimeExecutorParam,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct WasmRuntimeExecutorParam {
+    pub source_code_file_path: String,
+    pub wasm_file_path: String,
```

Comment:
The path to the wasm binary has a temporary folder, so please save it there.
I don't want it to be visible to the user.
I'll make it available for later reference from the context.

Reply:
I have thought about it a lot, but I think it would be better to create a temporary directory (tempfile crate), write once, and cache the binary in memory.